### PR TITLE
dep: Upgrade webpack

### PR DESCRIPTION
**Pull request by tahini**
See individual commits for details

### CODE REVIEW

File: packages/transition-frontend/webpack.config.js [hunk 1]
```diff
@@ -30,13 +30,13 @@ const entryFileName =  './lib/app-transition.js';
...
-  const isProduction = env === 'production';
   console.log('process.env.NODE_ENV', process.env.NODE_ENV);
 
-  const bundleFileName = isProduction ? `transition-${config.projectShortname}-bundle-${env}.[contenthash].js` : `transition-${config.projectShortname}-bundle-${env}.dev.js`;
+  const isProduction = process.env.NODE_ENV === 'production';
+
+  const bundleFileName = isProduction ? `transition-${config.projectShortname}-bundle-${process.env.NODE_ENV}.[contenthash].js` : `transition-${config.projectShortname}-bundle-${process.env.NODE_ENV}.dev.js`;
   const styleFileName = isProduction ? `transition-${config.projectShortname}-styles.[contenthash].css` : `transition-${config.projectShortname}-styles.dev.css`;
-  const htmlFileName = path.join(`index-${config.projectShortname}${env === 'test' ? `_${env}` : ''}.html`);
+  const htmlFileName = path.join(`index-${config.projectShortname}${process.env.NODE_ENV === 'test' ? `_${process.env.NODE_ENV}` : ''}.html`);
```
Comment (greenscientist) [inline packages/transition-frontend/webpack.config.js:39]: Can you add a comment to this line to explain what it does ? I want to have a confirmation that the logic is right.

File: packages/transition-frontend/webpack.config.js [hunk 2]
```diff
@@ -53,9 +53,13 @@ module.exports = (env) => {
   ];
 
   return {
+    stats: {
```
Comment (greenscientist) [inline packages/transition-frontend/webpack.config.js:56]: Comments for this?

**Review by greenscientist — CHANGES_REQUESTED** [review]
Just wanting a bit more comments.

File: packages/transition-frontend/webpack.config.js (hunk 2 above)
Comment (tahini) [reply packages/transition-frontend/webpack.config.js:56]: No clue! But ok, I,ll try to figure it out.... *grumble grumble*

Comment (greenscientist) [reply packages/transition-frontend/webpack.config.js:56]: I guess it's that: https://webpack.js.org/configuration/stats/   ??

File: packages/transition-frontend/webpack.config.js [hunk 3]
```diff
@@ -176,8 +179,10 @@ module.exports = (env) => {
       )
     ],
     resolve: {
+      mainFields: ['browser', 'main', 'module'],
```
Comment (greenscientist) [inline packages/transition-frontend/webpack.config.js:182]: I forgot to ask about the changes here, but at this point, I guess it's ok.

**Review by greenscientist — APPROVED** [review]

File: packages/transition-frontend/webpack.config.js (hunk 3 above)
Comment (tahini) [reply packages/transition-frontend/webpack.config.js:182]: I guess anyone who knows webpack better than us can figure it out...

I'll open an issue to revisit webpack config file anyway, I'm sure there's tons of optimization to make

**Pull request merged**